### PR TITLE
Annotate DbDataReader.GetSchemaTable as nullable

### PR DESCRIPTION
**Pull request by roji**
Reverts decision made in #509

/cc @ajcvickers @jeffhandley @stephentoub

### CODE REVIEW

**Review by jeffhandley — APPROVED** [review]
LGTM, but I recommend waiting for at least one more approval.

**Review by ajcvickers — APPROVED** [review]

**Pull request merged**
